applications/portfolio_manager/src/portfolio_manager/rebalance.py

```
@@ -0,0 +1,612 @@
+                "Failed to close position",
+                ticker=close_position["ticker"],
+                error=str(e),
+            )
+            close_results.append(
+                {
+                    "ticker": close_position["ticker"],
+                    "action": "close",
+                    "status": "failed",
+                    "error": str(e),
+                }
+            )
```

Ruff (0.15.6)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)